test/unit/puppetlabs/services/jruby_pool_manager/jruby_agents_test.clj

```diff
@@ -104,5 +36,21 @@
              pool-context (pool-manager-protocol/create-pool pool-manager-service config)]
          (jruby-core/flush-pool! pool-context)
          ; wait until the flush is complete
-         (jruby-testutils/timed-await (jruby-agents/get-pool-agent pool-context))
+         (jruby-testutils/wait-for-instances (jruby-core/get-pool pool-context) 1)
```

Comment:
Same comment here as in one of the tests above about how it seems like this step could complete before the pool has actually been flushed, depending upon timing.